pkg/network/state.go

```diff
@@ -1100,22 +1123,112 @@ type aggregateConnection struct {
 	count             uint32
 }
 
+type aggregationKey struct {
+	string
+	direction  ConnectionDirection
+	containers struct {
+		source, dest *intern.Value
+	}
+}
+
 type connectionAggregator struct {
-	conns    map[string][]*aggregateConnection
-	buf      []byte
-	dnsStats dns.StatsByKeyByNameByType
+	conns                       map[aggregationKey][]*aggregateConnection
+	buf                         []byte
+	dnsStats                    dns.StatsByKeyByNameByType
+	enablePortRollups           bool
+	processEventConsumerEnabled bool
 }
 
-func newConnectionAggregator(size int, dnsStats dns.StatsByKeyByNameByType) *connectionAggregator {
+func newConnectionAggregator(size int, enablePortRollups, processEventConsumerEnabled bool, dnsStats dns.StatsByKeyByNameByType) *connectionAggregator {
 	return &connectionAggregator{
-		conns:    make(map[string][]*aggregateConnection, size),
-		buf:      make([]byte, ConnectionByteKeyMaxLen),
-		dnsStats: dnsStats,
+		conns:                       make(map[aggregationKey][]*aggregateConnection, size),
+		buf:                         make([]byte, ConnectionByteKeyMaxLen),
+		dnsStats:                    dnsStats,
+		enablePortRollups:           enablePortRollups,
+		processEventConsumerEnabled: processEventConsumerEnabled,
+	}
+}
+
+func (a *connectionAggregator) key(c *ConnectionStats) (key aggregationKey, sportRolledUp, dportRolledUp bool) {
+	key.string = string(c.ByteKey(a.buf))
+	key.direction = c.Direction
+	if a.processEventConsumerEnabled {
+		key.containers.source = c.ContainerID.Source
+	}
+
+	if !a.enablePortRollups {
+		return key, false, false
+	}
+
+	// local resolution is done in system-probe if rollups
+	// are enabled, so add the destination container id to
+	// the key as well
+	key.containers.dest = c.ContainerID.Dest
+
+	isShortLived := c.IsClosed && (c.Duration > 0 && c.Duration < shortLivedConnectionThreshold)
+	sportRolledUp = c.Direction == OUTGOING
+	dportRolledUp = c.Direction == INCOMING
+
+	log.TraceFunc(func() string {
+		return fmt.Sprintf("type=%s isShortLived=%+v sportRolledUp=%+v", c.Type, isShortLived, sportRolledUp)
+	})
+	if !isShortLived ||
```

Comment:
Is there a reason or benefit for _not_ rolling up long-lived connections?
I can think of at least a few real-world scenarios where you could have expect to have a decent amount of long-lived connections matching the same (source, dest), like a connection pool between a LB and a web-server/reverse-proxy or a connection pool between something like a pgbouncer and a postgres DB

Reply:
See https://datadoghq.atlassian.net/wiki/spaces/NET/pages/3032351305/Agent+Connection+Roll+Ups#Challenges (2.)